### ISSUE TITLE
Using Simple API Gateway to deploy multiple stages

### ISSUE THREAD

**Issue by anush0247** [Miserlou/Zappa#628]
I am trying to deploy new environment using zappa deploy. 

I am getting an error like 

`botocore.exceptions.ClientError: An error occurred (BadRequestException) when calling the CreateRestApi operation: Maximum number of APIs has been reached.  Please contact AWS if you need additional APIs.`

When I raised a limit increase request in aws console

I got the reply like 

    1. If two APIs serve a similar business purpose, then merge the resources into one API.
    2. Use stages instead of creating separate dev/beta APIs.

are we going to get the support for deploying the multiple stages to same API Gateway API

**Comment by Miserlou** [comment]
You must have like.. at least 50 APIs on your account already?

**Comment by anush0247** [comment]
Yes, we have 60 APIs already deployed using zappa

**Comment by Miserlou** [comment]
Well hot damn!

I don't know what you're proposing Zappa do about this though. (I think you're the first person to hit this limit.) I would need to know a lot more about your system to comment properly here.

**Comment by anush0247** [comment]
Before the zappa we used the deploy multiple api environments to same api gateway api, as we came across the zappa we started deploying all our apis using zappa. Now we have reached our account max no of apis per account limit.

**Comment by kapeed2091** [comment]
Is there any provision in zappa wherein we can use the stage feature of apigateway?

Say a project named X has two branches 'dev' and 'prod', it would be of help if we can have same apigateway with each branch being pointed to a different stage (stages name can be same as the branch name) in apigateway. And I believe even AWS team has suggested the same.

**Comment by Miserlou** [comment]
Yeah, to isolate them was a design decision in Zappa from the get go, as having multiple stages on the same API gateway seemed like it added a big layer of complexity, although I didn't anticipate anybody having that many APIGWs on the same account. I also don't know how it interacts with the domains, that could be a headache too.

**Comment by kapeed2091** [comment]
You could probably provide another deployment option where user can create multiple stages.

**Comment by wobeng** [comment]
To accomplish this, I think we have to go from cloudformation to direct boto apigateway api calls and treat each config as stages and have one global config for main api

**Comment by nikbora** [comment]
+1 for this. We are also switching a lot of our internal and external APIs and worker style services to Zappa and will likely hit the API Gateway limit. Whilst I am sure we can talk to AWS about increasing the limit it would also be nice for Zappa to use APIGs staging capability so it's a bit more frugal.

**Comment by anush0247** [comment]
@nikbora we are using #663 to deploy our internal portals, Its working like a charm, this is pr is waiting from a long time

**Comment by deoresheetal** [comment]
Any progress on this for stages?

**Comment by anush0247** [comment]
@deoresheetal I will be updating the existing PR

**Comment by PeteW** [comment]
I would also like to see the use of stages put into play. 

Currently the documentation suggests that stages in the zappa_settings file translate into stages of a common API which seems like the intuitive behavior to a newcomer like me. I was surprised when I added a new stage to my zappa_settings deployed a new API instead of adding an API stage to the existing API

**Comment by curtismorte** [comment]
Much like @PeteW, I also discovered the exact same thing because the readme information makes it seem like you can use API Gateway Stages. However, the definition for Zappa is that a "stage" is in fact an independent API. If you review @wobeng's comment, you will understand why this is.
